secure/storage/vault/src/lib.rs

```diff
@@ -97,29 +97,28 @@ pub struct Client {
     agent: ureq::Agent,
     host: String,
     token: String,
-    tls_config: Option<Arc<rustls::ClientConfig>>,
+    tls_connector: Option<Arc<native_tls::TlsConnector>>,
 }
 
 impl Client {
     pub fn new(host: String, token: String, ca_certificate: Option<String>) -> Self {
-        let tls_config = if let Some(certificate) = ca_certificate {
-            let mut tls_config = rustls::ClientConfig::new();
+        let mut tls_builder = native_tls::TlsConnector::builder();
+        tls_builder.min_protocol_version(Some(native_tls::Protocol::Tlsv12));
+        if let Some(certificate) = ca_certificate {
             // First try the certificate as a DER encoded cert, then as a PEM, and then panic.
```

Comment:
it looks like you inverted this test

Reply:
Yeah, PEM is more common. I'll update the comment.